### PR TITLE
try switching from yasm to nasm

### LINKED ISSUE #30: use nasm?
it looks like `nasm` is the only supported assembler:

https://github.com/openssl/openssl/blob/9a3b5b766421b7ed9df859956d44fcbe52c2dd82/NOTES.WIN#L50-L54

is there a reason for using `yasm` here? Very likely yes (I imagine), but just wanted to check.

### PR DESCRIPTION
**Pull request by carlodri**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #30 

**NOTE:** I added myself as maintainer

### CODE REVIEW

**Comment by carlodri** [comment]
@conda-forge/openssl let me know if you spot any problems! LGTM.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,9 +26,8 @@ build:
 requirements:
   build:
     - {{ compiler('c') }}
-    - yasm               # [win32]
+    - nasm
     - perl
-    - make               # [unix]
```
Comment (jjhelmus) [inline recipe/meta.yaml]: can we keep make as a build dependency on unix?  The only reason it is not need is that make is included in the docker image, using the conda provided version is much preferred.

Comment (carlodri) [reply recipe/meta.yaml]: oh, sure, didn't know that, I erroneously supposed it was brought in by the compiler directive

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,9 +26,9 @@ build:
 requirements:
   build:
     - {{ compiler('c') }}
-    - yasm               # [win32]
+    - nasm
```
Comment (jakirkham) [inline recipe/meta.yaml]: Actually I think this is only a Windows requirement. In fact am now remembering this was [only needed/used on Windows 32-bit](  https://wiki.openssl.org/index.php/Compilation_and_Installation#W32_.2F_Windows_NT_-_Windows_9x ) as the selector notes.

There are some differences between support for 32-bit and 64-bit assembly on Windows. For example [there is no inline assembly support in VS for 64-bit]( https://docs.microsoft.com/en-us/cpp/build/intrinsics-and-inline-assembly?view=vs-2017 ). Not sure if this why [OpenSSL doesn't use assembly on Windows 64-bit]( https://wiki.openssl.org/index.php/Compilation_and_Installation#W64 ). Given this, maybe we should just drop the assembler requirement entirely. Thoughts?

Also sorry for the mix-up. 😞

Comment (jakirkham) [reply recipe/meta.yaml]: If we do decide to drop the assembler requirement, we can clean out some patches and other workarounds for YASM. ( https://github.com/conda-forge/openssl-feedstock/commit/cddd3f60d674258699a7853e99fb1a6d29cddef0 )

Comment (carlodri) [reply recipe/meta.yaml]: @jakirkham, thanks for your comments! I have already removed YASM patches and workarounds, and following your suggestion I have entirely removed the assembler requirement.

I think that what we are saying is valid for 1.0.2p, but it seems from the official docs that things have changed in the newer versions. In fact, in `openssl`'s `master` branch, `nasm` is stated as a strict requirement.

**Comment by carlodri** [comment]
@jjhelmus @jakirkham do you think this is ready?

**Comment by jjhelmus** [comment]
LGTM

**Pull request merged**